### PR TITLE
fix: JSON.stringify for bigint

### PR DESCRIPTION
**Pull request by chenyan-dfinity**
The original `toReadableString` function doesn't work when bigint is nested, e.g. `[BigInt(42)]`

### CODE REVIEW

File: packages/candid/src/idl.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export abstract class Type<T = any> {
   }
 
   public valueToString(x: T): string {
-    return JSON.stringify(x);
+    return JSON.stringify(x, (_, v) => (typeof v === 'bigint' ? v.toString : v));
```
Comment (kritzcreek) [inline packages/candid/src/idl.ts:178]:
```suggestion
    return JSON.stringify(x, (_, v) => (typeof v === 'bigint' ? v.toString() : v));
```

**Review by krpeacock — APPROVED** [review]
Nice, the JSON.stringify bug was a pain

**Comment by krpeacock** [comment]
@chenyan-dfinity are there any tests that explicitly cover this functionality?

**Comment by FloorLamp** [comment]
Oops, I didn't see this one. Test added in #474

**Comment by krpeacock** [comment]
Closing in favor of #474

**Pull request closed without merging**